cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -112,9 +111,10 @@ func NewDDLSink(
 
 // WriteDDLEvent writes a DDL event to the mysql database.
 func (m *DDLSink) WriteDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
-	if ddl.Type == timodel.ActionAddIndex && m.cfg.IsTiDB {
-		return m.asyncExecAddIndexDDLIfTimeout(ctx, ddl)
+	if m.shouldAsyncExecDDL(ddl) {
+		return m.asyncExecDDL(ctx, ddl)
 	}
+	m.waitAsynExecDone(ctx, ddl)
```

Comment:
It seems all DDLs need to go through this logic before they can be executed. Would this increase the time needed for a DDL to be replicated?

Reply:
Yes, it takes at least one additional request in TiDB scenario.

Reply:
sounds like we can add a variable to indicate that there might be one asynchronous running ddl ?

Reply:
It would be unacceptable if there is a huge amount of records in the downstream DDL table, let alone the time spent waiting here: https://github.com/pingcap/tiflow/pull/11476/files#diff-67001e77c3d88c6c47cfed312c5b1f5f36d2a404163aefb5da2939fce5051674R109

Reply:
Can multiple async DDLs be executed without waiting？

Reply:
We only [query one record](https://github.com/pingcap/tiflow/pull/11476/files#diff-67001e77c3d88c6c47cfed312c5b1f5f36d2a404163aefb5da2939fce5051674R36), so I understand that the overhead is fixed and does not depend on the size of the table.

Reply:
> It would be unacceptable if there is a huge amount of records in the downstream DDL table, let alone the time spent waiting here: https://github.com/pingcap/tiflow/pull/11476/files#diff-67001e77c3d88c6c47cfed312c5b1f5f36d2a404163aefb5da2939fce5051674R109

Got it, there is indeed a bug here that we should not be waited for 5s before the first check.